apps/fz_http/lib/fz_http/devices/device.ex

```diff
@@ -149,27 +113,66 @@ defmodule FzHttp.Devices.Device do
       greater_than_or_equal_to: 576,
       less_than_or_equal_to: 1500
     )
-    |> validate_length(:description, max: @description_max_length)
-    |> validate_ipv4_required()
-    |> validate_ipv6_required()
+    |> prepare_changes(fn changeset ->
+      changeset
+      |> maybe_put_default_ip(:ipv4)
+      |> maybe_put_default_ip(:ipv6)
+      |> validate_exclusion(:ipv4, [ipv4_address()])
+      |> validate_exclusion(:ipv6, [ipv6_address()])
+    end)
     |> unique_constraint(:ipv4)
     |> unique_constraint(:ipv6)
-    |> validate_exclusion(:ipv4, [ipv4_address()])
-    |> validate_exclusion(:ipv6, [ipv6_address()])
-    |> validate_in_network(:ipv4)
-    |> validate_in_network(:ipv6)
-    |> validate_no_mask(:ipv4)
-    |> validate_no_mask(:ipv6)
     |> unique_constraint(:public_key)
     |> unique_constraint([:user_id, :name])
   end
 
+  defp maybe_put_default_ip(changeset, field) do
+    if FzHttp.Config.fetch_env!(:fz_http, :"wireguard_#{field}_enabled") == true do
+      case fetch_field(changeset, field) do
+        {:data, nil} -> put_default_ip(changeset, field)
+        :error -> put_default_ip(changeset, field)
+        _ -> changeset
+      end
+      |> validate_required(field)
+    else
+      changeset
+    end
+  end
+
+  defp put_default_ip(changeset, field) do
+    cidr_string = FzHttp.Config.fetch_env!(:fz_http, :"wireguard_#{field}_network")
+    {:ok, cidr_inet} = EctoNetwork.INET.cast(cidr_string)
+    cidr = CIDR.parse(cidr_string)
+    offset = Enum.random(2..(cidr.hosts - 2))
+
+    {:ok, gateway_address} =
+      FzHttp.Config.fetch_env!(:fz_http, :"wireguard_#{field}_address")
+      |> EctoNetwork.INET.cast()
+
+    Devices.Device.Query.next_available_address(cidr_inet, offset, [gateway_address])
+    |> FzHttp.Repo.one()
+    |> case do
+      nil -> add_error(changeset, :base, "CIDR #{cidr} is exhausted")
```

Comment:
Would it be worth tracking when this happens? Maybe not in the schema module, but perhaps in the `FzHttp.Telemetry.ping` data we can add a `cidr_size()` data point to track how much of the configured range is being used.

```suggestion
      nil ->
        FzHttp.Telemetry.cidr_exhausted(cidr)
        add_error(changeset, :base, "CIDR #{cidr} is exhausted")
```

Reply:
We can do that, but it won't give us any value right now, we already know how many devices are added. In the future, I would love to see us building an issue tracker for potential customers (we can build a system that detects potential issues and notifies both them and us).

Reply:
Moreso to get an idea how much of the configured range is being used. But if we're switching to CGNAT anyway it'd be pretty useless agreed due to the negligible change of collision.